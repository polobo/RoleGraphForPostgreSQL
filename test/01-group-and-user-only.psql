-- Demonstrate what happens when user and group roles are created during the cr and su passes
-- Specifically, under superuser there are no listed administration relationships while
-- due to createrole_self_grant and the nature of create role there are admin, set, and inherit
-- grants from admin to each of the created roles.

SET createrole_self_grant TO set, inherit;
SELECT format('CREATE ROLE u1_alice_%s LOGIN;', :'ctx') \gexec
SELECT format('CREATE ROLE g1_service_advisor_%s NOLOGIN;', :'ctx') \gexec
