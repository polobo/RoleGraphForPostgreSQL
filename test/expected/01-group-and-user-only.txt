                                                                            Test Result for 01-group-and-user-only
        rolname        |             administration              |         memberof_users          |              memberof_groups              |         member_users         | member_groups 
-----------------------+-----------------------------------------+---------------------------------+-------------------------------------------+------------------------------+---------------
 g1_service_advisor_cr | by cr_admin from superuser              |                                 |                                           | cr_admin from cr_admin (s,i) | 
 g1_service_advisor_su |                                         |                                 |                                           |                              | 
 cr_admin              | Create Role                            +| u1_alice_cr from cr_admin (s,i) | g1_service_advisor_cr from cr_admin (s,i) |                              | 
                       | of u1_alice_cr from superuser          +|                                 |                                           |                              | 
                       | of g1_service_advisor_cr from superuser |                                 |                                           |                              | 
 u1_alice_cr           | by cr_admin from superuser              |                                 |                                           | cr_admin from cr_admin (s,i) | 
 u1_alice_su           |                                         |                                 |                                           |                              | 
(5 rows)

